nucliadb/src/nucliadb/writer/api/v1/export_import.py

```diff
@@ -75,6 +84,55 @@ async def start_kb_export_endpoint(request: Request, kbid: str):
         return CreateExportResponse(export_id=export_id)
 
 
+@only_for_onprem
+@api.post(
+    f"/{KBS_PREFIX}/import",
+    summary="Create a KB from an export and import its content",
+    tags=["Knowledge Boxes"],
+    response_model=NewImportedKbResponse,
+    openapi_extra={"x-hidden-operation": True},
+)
+@requires_one([NucliaDBRoles.MANAGER, NucliaDBRoles.WRITER])
+@version(1)
+async def kb_create_and_import_endpoint(request: Request):
+    context = get_app_context(request.app)
+
+    # Read stream and parse learning configuration
+    stream_reader = ExportStreamReader(request.stream())
+    learning_config, leftover_bytes = await stream_reader.maybe_read_learning_config()
+    if learning_config is None:
+        return HTTPClientError(
+            status_code=400,
+            detail="Trying to import an export missing learning config. Try using import on an existing KB or use a newer export",
+        )
+
+    # Create a KB with the import learning config
+
+    now = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    import_kb_config = KnowledgeBoxConfig(
+        title=f"Imported KB - {now}",
+        learning_configuration=learning_config.dict(),
+    )
+    kbid, slug = await create_kb(import_kb_config)
+
+    # Import contents to the new KB
+
+    async def stream_with_leftovers(leftovers: bytes, stream: AsyncGenerator[bytes, None]):
+        if len(leftovers) > 0:
+            yield leftovers
+        async for chunk in stream:
+            yield chunk
+
+    await importer.import_kb(
```

Comment:
I think we should not be importing synchronously here, and we should return the import_id for the client to check the status

Reply:
Why not? I'm following the pattern of the regular import where standalone does this and hosted uploads to a bucket and then imports